script.js

```diff
@@ -1,4 +1,120 @@
+// INPUT VALIDATION NEEDS WORK... AAAARGH!
+
+// GLOBAL VARIABLES
+
+var gameModeDiceRoll = "dice roll";
+var gameModeChooseOrder = "player choice";
+var gameModeCompareScores = "compare scores";
+var gameMode = gameModeDiceRoll;
+var currentPlayer = 1;
+
+// ARRAYS
+var currentPlayerRolls = [];
+var allScores = [];
+
+//helper functions
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var getPlayerHand = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter += 1;
+  }
+  return `Welcome Player ${currentPlayer}. <br><br> You rolled ${currentPlayerRolls[0]} for Dice 1 and ${currentPlayerRolls[1]} for Dice 2. <br><br> Please choose the order of the dice by typing 1 or 2 and clicking submit to indicate which dice you want to use as your first number.<br><br> Remember, the larger your score, the greater your chances of winning.`;
+};
+
+var choosePlayerScoreOrder = function (playerSelection) {
+  var playerScore = "";
+  if (playerSelection == "1") {
+    playerScore = String(currentPlayerRolls[0]) + String(currentPlayerRolls[1]);
+    allScores.push(playerScore);
+    myOutputValue = `Player ${currentPlayer}, you have selected ${playerSelection} and your score is ${playerScore}. `;
+    currentPlayerRolls = [];
+    gameMode = gameModeDiceRoll;
+    return myOutputValue;
+  } else if (playerSelection == "2") {
+    playerScore = String(currentPlayerRolls[1]) + String(currentPlayerRolls[0]);
+    allScores.push(playerScore);
+    myOutputValue = `Player ${currentPlayer}, you have selected ${playerSelection} and your score is ${playerScore}. `;
+    currentPlayerRolls = [];
+    gameMode = gameModeDiceRoll;
+    return myOutputValue;
+  } else {
+    myOutputValue = `Your input was invalid, please enter either "1" to use ${currentPlayerRolls[0]} or "2" to use ${currentPlayerRolls[1]} as the first number of the dice score.`;
+  }
```

Comment:
Good effort for trying to implement some form of input validation! However, I'm sure you've noticed that the code continues to run despite this. The reason for this is because the code does not breakout of its sequence, and continues from line 73 and onwards. Player is now "2" and gameMode is still "player choice". Therefore the next time you hit the submit button, the code jumps to line 86. 

What you need to do is to try and breakout of the sequence. You can consider including an 'if' statement at line 73 which returns a value if the input was invalid.